sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/package.scala

```diff
@@ -162,7 +162,8 @@ package object util {
   def toCommentSafeString(str: String): String = {
     val len = math.min(str.length, 128)
     val suffix = if (str.length > len) "..." else ""
-    str.substring(0, len).replace("*/", "\\*\\/").replace("\\u", "\\\\u") + suffix
+    str.substring(0, len).replace("*/", "\\*\\/")
+      .replaceAll("(^|[^\\\\])(\\\\(\\\\\\\\)*u)", "$1\\\\$2") + suffix
```

Comment:
Is the implementation of `org.apache.commons.lang3.StringEscapeUtils.escapeJava(...)` sufficient to cover this case?